db/dcrpg/tables.go

```diff
@@ -12,22 +12,31 @@ import (
 	"github.com/decred/dcrdata/db/dcrpg/v4/internal"
 )
 
-var createTableStatements = map[string]string{
-	"meta":           internal.CreateMetaTable,
-	"blocks":         internal.CreateBlockTable,
-	"transactions":   internal.CreateTransactionTable,
-	"vins":           internal.CreateVinTable,
-	"vouts":          internal.CreateVoutTable,
-	"block_chain":    internal.CreateBlockPrevNextTable,
-	"addresses":      internal.CreateAddressTable,
-	"tickets":        internal.CreateTicketsTable,
-	"votes":          internal.CreateVotesTable,
-	"misses":         internal.CreateMissesTable,
-	"agendas":        internal.CreateAgendasTable,
-	"agenda_votes":   internal.CreateAgendaVotesTable,
-	"testing":        internal.CreateTestingTable,
-	"proposals":      internal.CreateProposalsTable,
-	"proposal_votes": internal.CreateProposalVotesTable,
+var createTableStatements = [][2]string{
+	{"meta", internal.CreateMetaTable},
+	{"blocks", internal.CreateBlockTable},
+	{"transactions", internal.CreateTransactionTable},
+	{"vins", internal.CreateVinTable},
+	{"vouts", internal.CreateVoutTable},
+	{"block_chain", internal.CreateBlockPrevNextTable},
+	{"addresses", internal.CreateAddressTable},
+	{"tickets", internal.CreateTicketsTable},
+	{"votes", internal.CreateVotesTable},
+	{"misses", internal.CreateMissesTable},
+	{"agendas", internal.CreateAgendasTable},
+	{"agenda_votes", internal.CreateAgendaVotesTable},
+	{"testing", internal.CreateTestingTable},
+	{"proposals", internal.CreateProposalsTable},
+	{"proposal_votes", internal.CreateProposalVotesTable},
+	{"stats", internal.CreateStatsTable},
+}
+
+func createTableMap() map[string]string {
+	m := make(map[string]string, len(createTableStatements))
+	for _, pair := range createTableStatements {
+		m[pair[0]] = pair[1]
```

Comment:
Just curious why these changes.

Reply:
Since `stats` has a foreign key constraint, I needed to create the tables in order.

Reply:
Ahhh, makes sense.